src/main/java/uk/gov/justice/digital/hmpps/csr/api/dto/ShiftNotificationDto.kt

```diff
@@ -0,0 +1,66 @@
+package uk.gov.justice.digital.hmpps.csr.api.dto
+
+import java.util.*
+import com.fasterxml.jackson.annotation.JsonCreator
+import com.fasterxml.jackson.annotation.JsonProperty
+import io.swagger.annotations.ApiModel
+import io.swagger.annotations.ApiModelProperty
+import uk.gov.justice.digital.hmpps.csr.api.domain.ActionType
+import uk.gov.justice.digital.hmpps.csr.api.domain.ShiftType
+import uk.gov.justice.digital.hmpps.csr.api.model.ShiftNotification
+import java.time.Clock
+import java.time.LocalDate
+import java.time.LocalDateTime
+
+@ApiModel(description = "ShiftNotification")
+data class ShiftNotificationDto @JsonCreator constructor(
+        @ApiModelProperty(required = true, value = "User unique identifier", position = 1, example = "AB102CD")
+        @JsonProperty("quantumId")
+        val quantumId: String,
+
+        @ApiModelProperty(required = true, value = "When the shift notification was created", position = 2, example = "2020-04-20T17:45:55")
+        @JsonProperty("createdAt")
+        val createdAt: LocalDateTime,
+
+        @ApiModelProperty(required = true, value = "Staff ID number", position = 3, example = "123")
+        @JsonProperty("staffId")
+        val staffId: Int,
+
+        @ApiModelProperty(required = true, value = "Date of shift", position = 4, example = "2020-08-22")
+        @JsonProperty("shiftDate")
+        val shiftDate: LocalDate,
+
+        @ApiModelProperty(required = true, value = "Date shift was last modified", position = 5, example = "2020-08-22T09:45:55")
+        @JsonProperty("lastModified")
+        val lastModified: LocalDateTime,
+
+        @ApiModelProperty(required = true, value = "Type of shift", position = 6, example = "SHIFT")
+        @JsonProperty("shiftType")
+        val shiftType: String,
+
+        @ApiModelProperty(required = true, value = "Action type of notification", position = 3, example = "EDIT")
+        @JsonProperty("actionType")
+        val actionType: String
+) {
+        companion object {
+
+                fun from(shiftNotifications: Collection<ShiftNotification>): Collection<ShiftNotificationDto> {
+                        return shiftNotifications.map {
+                                from(it)
+                        }
+                }
+
+                fun from(it: ShiftNotification): ShiftNotificationDto {
+                        return ShiftNotificationDto(
+                                it.quantumId,
+                                it.createdAt,
+                                it.staffId,
+                                it.shiftDate,
+                                it.lastModified,
+                                ShiftType.fromInt(it.shiftType)!!.shiftType,
```

Comment:
Come to think of it, I think the `!!` is in the correct place here. The enum shouldn't enforce a default as it may be useful in other methods/contexts. It's this function that specifically wants to overwrite the potential null value

Reply:
We're saying it will never be null, but it can be null because internally I know we're just getting map[value]... we should have a default here with ?: or preferably in ShiftNotification or the service?

Reply:
Maybe I am being too strict with data here, but I feel like putting a default in is swallowing the error. If this returns null that means there's been a breaking change with the data source and we need to know what's happened.

Reply:
I guess I just see 'edit' as meaning 'something changed' as a fall back. what's going to happen here otherwise is the entire call will fail with a NPE and we'll have to contact CSR or even the prison detailer to correct to the mistake in a database we don't control.

Reply:
Yeah okay, I am less used to working with dbs I don't have access to. Do you think it's overkill to log that the overwright is happening? Some type of tracibility would make me feel a bit happier

Reply:
log.warn territory then